app/controllers/bookingsync_portal/admin/remote_accounts_controller.rb

```diff
@@ -9,6 +9,7 @@ def new
 
       def create
         @remote_account = scope.create(params_remote_account)
+        synchronize_bookingsync_source_id if current_account.synced_source_id.blank?
```

Comment:
would extract it (imho easiest to account model, `ensure_source_exists!` that would check if account has source in core and do all this logic, move from controller to model, or if you are into services ghetto as @Azdaroth make it a service :P)